.github/workflows/cd.yml

```diff
@@ -0,0 +1,136 @@
+name: CD
+
+on:
+  push:
+    tags-ignore:
+      - "*.dev0"
+  pull_request:
+    branches:
+      - develop
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  Build-Windows-x64:
+    runs-on: windows-latest
+
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.9"
+          architecture: "x64"
+
+      - name: Cache pip
+        uses: actions/cache@v2
+        with:
+          path: '%LOCALAPPDATA%\pip\Cache'
+          # Look to see if there is a cache hit for the corresponding requirements file
+          key: ${{ runner.os }}-pip-${{ hashFiles('tox.ini', 'setup.py', 'requirements*.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+            ${{ runner.os }}-
+
+      - name: Install dependencies
+        run: |
+          $WebClient = New-Object System.Net.WebClient
+          $WebClient.DownloadFile("https://ci.appveyor.com/api/buildjobs/vggwhkyrv0cpoewf/artifacts/gvsbuild-vs16-x64.zip","C:\GTK.zip")
+          7z x C:\GTK.zip -oc:\GTK
+          $env:Path = "C:\GTK\release;C:\GTK\release\bin;C:\GTK\release\lib;$env:Path"
+          $Env:Path
+          python -m pip install --upgrade pip wheel
+          pip install C:/GTK/release/python/pycairo-1.20.1-cp37-cp37m-win_amd64.whl
+          pip install C:/GTK/release/python/PyGObject-3.42.0-cp37-cp37m-win_amd64.whl
+          python -m pip install libtorrent==1.2.*
+          pip install -r requirements.txt
+          pip install pyinstaller
+
+      - name: Install Deluge
+        run: |
+          python setup.py clean
+          python setup.py build
+          python setup.py install
+
+      - name: Freeze Deluge
+        run: |
+          gci env:* | sort-object name
+          $env:Path = "C:\GTK\release;C:\GTK\release\bin;C:\GTK\release\lib;$env:Path"
+          pyinstaller --clean $env:GITHUB_WORKSPACE\packaging\win\delugewin.spec --distpath $env:GITHUB_WORKSPACE\packaging\win\freeze
+
+      - name: Make Deluge Installer
+        run: |
+          Set-Location -Path $env:GITHUB_WORKSPACE\packaging\win
+          python delugensis.py
+          makensis /Darch=x64 deluge-win-installer.nsi
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: my-artifact
+          path: packaging/win/*.exe
+
+  Build-Windows-x86:
```

Comment:
Can we use [strategy matrix](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#using-environment-variables-in-a-matrix) to deduplicate the Build-Windows steps.

So perhaps set a `gvsbuild` for download link, `arch` key set to `x86` or `x64` and `wheel_arch` to `win32` or `amd64`  etc...